retis/src/collect/collect.rs

```diff
@@ -317,17 +317,37 @@ impl Collectors {
         }
         Self::setup_filters(self.probes.builder_mut()?, collect)?;
 
-        // If probe_stack is on and user hasn't provided a starting point, use
-        // skb:consume_skb & skb:kfree_skb.
-        if collect.probe_stack && collect.probes.is_empty() {
-            self.probes
-                .builder_mut()?
-                .register_probe(Probe::raw_tracepoint(Symbol::from_name(
-                    "skb:consume_skb",
-                )?)?)?;
-            self.probes
-                .builder_mut()?
-                .register_probe(Probe::raw_tracepoint(Symbol::from_name("skb:kfree_skb")?)?)?;
+        // If auto_mode is on and no probe is given, find the right set
+        // automagically.
+        if self.auto_mode && collect.probes.is_empty() {
+            let mut probes = if collect.probe_stack {
+                // If --probe-stack is used, use skb:consume_skb & skb:kfree_skb
+                // as a starting point (these should capture most if not all of
+                // the packets and help moving up the stack).
+                vec![
+                    Probe::raw_tracepoint(Symbol::from_name("skb:consume_skb")?)?,
+                    Probe::raw_tracepoint(Symbol::from_name("skb:kfree_skb")?)?,
```

Comment:
wouldn't `skb_drop` collector enable kfree_skb anyways?

Reply:
That's right. I think this should still be explicitly set here:
- It makes it intentional as part as the default probe set for `--probe-stack`.
- The two probes (here & `skb-drop` collector) do not have the same intent.
- The two probes will be merged by the manager before being installed.